### PR TITLE
 [Backport stable/8.2] detect health issue when partition is stuck in a transition

### PR DESCRIPTION
**Pull request by rodrigo-lourenco-lopes**
## Description

Backports issue #13042

### CODE REVIEW

File: broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/PartitionTransitionImpl.java [hunk 1]
```diff
@@ -31,6 +33,7 @@ public final class PartitionTransitionImpl implements PartitionTransition {
   // transient state - to keep track of the current transitions
   private PartitionTransitionProcess currentTransition;
   private ActorFuture<Void> currentTransitionFuture;
+  private final Duration transitionStepTimeout = Duration.ofSeconds(60);
```
Comment (oleschoenburg) [inline broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/PartitionTransitionImpl.java:36]: :x: This is unused

**Review by lenaschoenburg — APPROVED** [review]
:clap: 

Please remove the unused variable before merging

**Comment by rodrigo-lourenco-lopes** [comment]
bors merge

**Comment by oleschoenburg** [comment]
bors retry

**Pull request merged**